<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LinearFilterDialog</class>
 <widget class="QDialog" name="LinearFilterDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>618</width>
    <height>380</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>線形フィルタ</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resource.qrc">
    <normaloff>:/icon/icon/filter.png</normaloff>:/icon/icon/filter.png</iconset>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>フィルタ係数</string>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>行列(&amp;M):</string>
          </property>
          <property name="buddy">
           <cstring>textEditM</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPlainTextEdit" name="textEditM">
          <property name="whatsThis">
           <string>フィルタの係数行列を指定します。
行列は 奇数行 × 奇数列の整数の行列です。
スペースで列を、改行で行を表してください。</string>
          </property>
          <property name="plainText">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>分母(&amp;D):</string>
          </property>
          <property name="buddy">
           <cstring>lineEditD</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="lineEditD">
          <property name="whatsThis">
           <string>行列の各々の要素を分母で割ります。分母を整数で指定してください。</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>後処理</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QRadioButton" name="radioButtonZ">
          <property name="text">
           <string>負の値を 0 にする(&amp;Z)</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButtonA">
          <property name="text">
           <string>負の値を絶対値にする(&amp;A)</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>例:</string>
       </property>
       <property name="buddy">
        <cstring>listWidgetFilter</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="listWidgetFilter">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>LinearFilterDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>164</x>
     <y>363</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>LinearFilterDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>164</x>
     <y>363</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textEditM</sender>
   <signal>textChanged()</signal>
   <receiver>LinearFilterDialog</receiver>
   <slot>setAutoByMatrix()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>1</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listWidgetFilter</sender>
   <signal>itemClicked(QListWidgetItem*)</signal>
   <receiver>LinearFilterDialog</receiver>
   <slot>setExampleFilter(QListWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>506</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>618</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listWidgetFilter</sender>
   <signal>itemDoubleClicked(QListWidgetItem*)</signal>
   <receiver>LinearFilterDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>516</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>619</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setAutoByMatrix()</slot>
  <slot>setExampleFilter(QListWidgetItem*)</slot>
 </slots>
</ui>
